collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java

```diff
@@ -170,7 +178,42 @@ public boolean checkAccountPermission() {
     }
 
     public String getSelectedAccount() {
-        return (String) preferences.get(PreferenceKeys.KEY_SELECTED_GOOGLE_ACCOUNT);
+        Account[] googleAccounts = credential.getAllAccounts();
+        String account = (String) preferences.get(PreferenceKeys.KEY_SELECTED_GOOGLE_ACCOUNT);
+
+        if (account == null) {
```

Comment:
This condition is always `false`. The default value for this key is "" (blank) and not null
Line 106 : `PreferencesKeys.java`